File: cpp/GOL.cpp
#include <iostream>
#include <stdlib.h>
#include <windows.h>
#include <stdio.h>
#include <time.h>
#include "GOL.h"

namespace std {

void GOL::fill(int p[][2], int s){
	for (int i=0; i<s; i++)
		w[p[i][0]][p[i][1]]=1;
}


void GOL::show (){
	for (int i=0; i<d; i++){
		for (int j=0; j<d; j++)
			if (w[i][j]==1)
				cout << "#";
			else
				cout << " ";

		cout << endl;
	}
};


void GOL::next(){

		int w2[d][d] = {};

	for (int r=0; r<d; r++)
		for (int c=0; c<d; c++){
			int n=0;
			if (r>0){
				if (c>0)
					n+=w[r-1][c-1];

				n+=w[r-1][c];
				if (c<d-1)
					n+=w[r-1][c+1];

			}
			if (c>0)
				n+=w[r][c-1];

			if (c<d-1)
				n+=w[r][c+1];

			if (r<d-1){
				if (c>0)
					n+=w[r+1][c-1];

				n+=w[r+1][c];
				if (c<d-1)
					n+=w[r+1][c+1];

			}

			if (n==3 || (n==2 && w[r][c]==1))
				w2[r][c]=1;
		}



	for (int i=0; i<d; i++)
		for (int j=0; j<d; j++)
			w[i][j]=w2[i][j];

};


GOL::GOL() {
	
}

GOL::~GOL() {
	
}

}

File: python/gol.py
import os
import time

class GOL:
	def __init__(self):
		self.d=25
		self.w = [[0 for j in range(self.d)] for i in range(self.d)]
	
	def fill(self, fillArray):
		for f in fillArray:
			self.w[f[0]][f[1]]=1
			

	def show(self):
		time.sleep(0.1) 
		os.system('cls')
		for r in range(len(self.w)):
			disp=""
			for c in range(len(self.w[r])):
				disp+="#" if self.w[r][c]==1 else " "
			print(disp)			

	def next(self):
		w=self.w
		d=self.d
		w2 = [[0 for j in range(d)] for i in range(d)]

		for r in range(len(w)):
			for c in range(len(w[r])):
				n=0
				if r>0:
					if c>0:
						n+=w[r-1][c-1]
					n+=w[r-1][c]
					if c<d-1:
						n+=w[r-1][c+1]
				if c>0:
					n+=w[r][c-1]
				if c<d-1:
					n+=w[r][c+1]
				if r<d-1:
					if c>0:
						n+=w[r+1][c-1]
					n+=w[r+1][c]
					if c<d-1:
						n+=w[r+1][c+1]
				#compute next generation
				if(n==3 or (n==2 and w[r][c]==1)):
					w2[r][c]=1 
		self.w=w2